Drive/rover_drive_system.hpp

```diff
@@ -34,63 +44,130 @@ class RoverDriveSystem
 
   void Enable(bool enable = true)
   {
-    rover_is_operational_ = enable;
+    mission_control_data_.is_operational = true;
     left_wheel_.Enable(enable);
     right_wheel_.Enable(enable);
     back_wheel_.Enable(enable);
-    SetMode('S');
+    SetMode();
   }
 
-  /// Main function for handling all the rover drive system functionality. Might
-  /// not need to take in params since maybe GetMissionControlData() will be
-  /// responsible for grabbing commands and saving them within the
-  /// RoverDriveSystem class.
-  void Move(char mode,
-            units::angle::degree_t rotation_angle,
-            units::angular_velocity::revolutions_per_minute_t speed)
+  // Retrieves /drive Mission Control commands.
+  /// @return returns true if connection is established from mission control
+  bool GetMissionControlData()
   {
-    if (rover_is_operational_)
+    // GET /drive JSON value ?
+    if (true)  // change 'true' to something for localhost connection found
+    {
+      char response[] = {};
+      ParseMissionControlData(response);
+      Move(mission_control_data_.drive_mode,
+           mission_control_data_.rotation_angle, mission_control_data_.speed);
+      return true;
+    }
+    sjsu::LogError("Unable to reach mission control server");
+    return false;
+  }
+
+  /// Main function for handling all the rover drive system functionality.
+  /// @param mode Updates drive mode if new mode is different from current.
+  /// @param rotation_angle adjusts wheel position depending on mode.
+  /// @param speed adjusts the movement speed of the rover.
+  void Move(char mode, float rotation_angle, float speed)
+  {
+    if (mission_control_data_.is_operational)
     {
-      // checks if the mode is different than what is currently running
       if (mode != static_cast<char>(current_mode_))
       {
         SetMode(mode);
       }
-      HandleModeMovement(rotation_angle, speed);
+      HandleRoverMovement(rotation_angle, speed);
```

Comment:
It might be a good idea to not attempt to move the rover after getting a new mode, but to request again what the new movement commands are. This is more of a safety I remember doing before.

Reply:
Made it an if/else statement to resolve issue